pandas/core/generic.py

```diff
@@ -6341,7 +6341,7 @@ def fillna(
                         continue
                     obj = result[k]
                     downcast_k = downcast if not is_dict else downcast.get(k)
-                    obj.fillna(v, limit=limit, inplace=True, downcast=downcast_k)
+                    result[k] = obj.fillna(v, limit=limit, downcast=downcast_k)
```

Comment:
We could do ``result[k].fillna(v, limit=limit, downcast=downcast_k)``

Is more clear

Reply:
Good call, have changed